New component: Sumo Logic extension (#29603)
open-telemetry/opentelemetry-collector-contrib: extension/sumologicextension/README.md

# Sumo Logic Extension

> **Stability level**: Beta

Reviewer: doesn't match metadata.yaml

Author: Fixed

| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Aextension%2Fsumologic%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Aextension%2Fsumologic) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Aextension%2Fsumologic%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Aextension%2Fsumologic) |
| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@astencel-sumo](https://www.github.com/astencel-sumo), [@sumo-drosiek](https://www.github.com/sumo-drosiek), [@swiatekm-sumo](https://www.github.com/swiatekm-sumo) |

[development]: https://github.com/open-telemetry/opentelemetry-collector#development
[sumo]: https://github.com/SumoLogic/sumologic-otel-collector
<!-- end autogenerated section -->

> This extension is to be used as part of Sumo Logic collector in conjunction with

Reviewer 2: ```suggestion
This extension is to be used in conjunction with
```

- `api_base_url`: base API URL that will be used for creating API requests,
  see [API URLs](#api-urls) details
  (default: `https://open-collectors.sumologic.com`)
- `heartbeat_interval`: interval that will be used for sending heartbeats
  (default: `15s`)
- `collector_credentials_directory`: directory where state files with registration
  info will be stored after successful collector registration
  (default: `$HOME/.sumologic-otel-collector`)
- `clobber`: defines whether to delete any existing collector with the same name. See [here][clobber] for more information.
- `force_registration`: defines whether to force registration every time the
  collector starts.
  This will cause the collector to not look at the locally stored credentials
  and to always reach out to API to register itself. (default: `false`)

  **NOTE**: if clobber is unset (default) then setting this to true will create
  a new collector (with new unique name) on Sumo UI on every collector start
  and create a new one upon registration.
- `ephemeral`: defines whether the collector will be deleted after 12 hours
  of inactivity (default: `false`)

> - `time_zone`: defines the time zone of the collector. For a list of all possible
>   values, refer to the `TZ database name` or `Time zone abbreviation` columns in
>   https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List

Reviewer 2: ```suggestion
- `time_zone`: defines the time zone of the collector, for example "Americas/Los_Angeles".
  For a list of all possible values, refer to the `TZ identifier` column in
  https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
```

- `backoff`: defines backoff mechanism for retry in case of failed registration.
  [Exponential algorithm](https://pkg.go.dev/github.com/cenkalti/backoff/v4#ExponentialBackOff) is being used.
  - `initial_interval` - initial interval of backoff (default: `500ms`)
  - `max_interval` - maximum interval of backoff (default: `1m`)
  - `max_elapsed_time` - time after which registration fails definitely (default: `15m`)

[credentials_help]: https://help.sumologic.com/docs/manage/security/installation-tokens
[fields_help]: https://help.sumologic.com/docs/manage/fields
[clobber]: https://help.sumologic.com/docs/send-data/installed-collectors/collector-installation-reference/force-collectors-name-clobber/

## Example Config

```yaml
extensions:
  sumologic:
    installation_token: <token>
    collector_name: my_collector

>     time_zone: CET

Reviewer 2: ```suggestion
    time_zone: Europe/Warsaw
```
